### ISSUE TITLE
Transform function

### ISSUE THREAD

**Issue by hajarhomayouni** [nicodv/kmodes#91]
Does K-prototypes implement the transform function of k-means, which returns a distance matrix?

**Comment by nicodv** [comment]
Can you be more specific, @hajarhomayouni ?

**Comment by hajarhomayouni** [comment]
Sklearn K-means has a [fit_transform ](http://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html#sklearn.cluster.KMeans.fit_transform) function that transforms X to a cluster-distance space.  
Is there a same function for k-prototypes?

**Comment by nicodv** [comment]
No, there's no `transform` method currently. `KPrototypes` would have to return both a dissimilarity matrix and a distance matrix then. Wouldn't be too hard to implement, I think.

What's your use case?

**Comment by hajarhomayouni** [comment]
I am trying to find local outliers in each cluster. So, I needed to calculate the distance between each data point to the corresponding cluster center. 
The transform function would give me the entire matrix, but now I need to calculate dissimilarity for each data point, which is not hard to implement. 

Thanks.
